src/content/1.5/code/agda/full.agda

```diff
@@ -0,0 +1,195 @@
+absurd ()
+
+{- This is the familiar "ex falso quodlibet" principle: from a false premise
+anything follows.                                                                -}
+
+{- SECTION 5.2: TERMINAL OBJECT --------------------------------------------------}
+{- The *terminal object* is the object with exactly one morphism coming to it from
+any other object in the category. It is unique up to isomorphism (when it exists).
+In the category of sets and functions, the terminal object is the singleton.
+In Agda, we represent the singleton by the unit type ⊤.                          -}
+unit : A → ⊤
+unit _ = tt
+
+{- Here's another morphism (in Set) from any object to the two element set Bool: -}
+yes : A → Bool
+yes _ = true
+
+{- but Bool does not meet the uniqueness criterion of terminal object, as there
+are other morphisms from every set to Bool; e.g.,                                -}
+no : A → Bool
+no _ = false
+
+{- SECTION 5.4: ISOMORPHISMS -----------------------------------------------------}
+{- Morphism 𝑔 is the inverse of morphism 𝑓 if their composition is the identity
+morphism. These are actually two equations because there are two ways of composing
+two morphisms:                                                                   -}
+_ : f ∘ g ≡ id
+_ = {!!}   -- We leave an unfilled hole where a proof would normally go;
+           -- here a proof is impossible since we don't know anything about f or g.
+_ : g ∘ f ≡ id
+_ = {!!}
+
+{- SECTION 5.5: PRODUCTS ---------------------------------------------------------}
+{- The *product* of two objects `A` and `B` is an object `A × B` together with two
+morphisms `π₁ : A × B → A` and `π₂ : A × B → B` such that for any object `C`
+and morphisms `f : C → A`, `g : C → B` there is a unique morphism `h : C → A × B`
+such that `f ≡ π₁ ∘ h` and `g ≡ π₂ ∘ h`.                                         -}
+
+------------------------------------------------------------------------
+-- Syntax for Σ-types (see Data.Sum.Base of the Agda standard library)
+open import Agda.Builtin.Sigma hiding (module Σ)
+module Σ = Agda.Builtin.Sigma.Σ
+Σ-syntax = Σ
+syntax Σ-syntax A (λ x → B) = Σ[ x ∈ A ] B
+infix 2 Σ-syntax
+
+{- Let's try to define a pattern of objects and morphisms in the category of sets
+that will lead us to the construction of a product of two sets, `A` and `B`. This
+pattern consists of an object `C` and two morphisms `p` and `q` connecting it to
+`A` and `B`, respectively:                                                       -}
+𝑝 : C → A
+𝑝 = {!!}
+𝑞 : C → B
+𝑞 = {!!}
+
+{- All `C`s that fit this pattern will be considered candidates for the product.
+There may be lots of them. For instance, let's pick, as our constituents, two
+Haskell types, `Int` and `Bool`, and get a sampling of candidates for their
+product. Here's one: `Int`. Can `Int` be considered a candidate for the product of
+`Int` and `Bool`? Yes, it can---and here are its projections:                    -}
+p₁ : Int → Int
```

Comment:
I used nested modules to hide conflicting definitions like so:
```agda
module snippet10 where
  p : Int → Int
  p x = x
  q : Int → Bool
  q _ = true
```
It depends do we actually want at some point verify e.g. on CI if all snippets are contained in summary file like full.agda.

I think it would be nice to have. But I am not 100% sure. WDYT?

Reply:
Good idea... it would be nice, though I'm not sure it's worth the extra effort it would require.  It seems to me that, in the original text, little or no attention was given to ensuring that the snippets are compilable Haskell code.